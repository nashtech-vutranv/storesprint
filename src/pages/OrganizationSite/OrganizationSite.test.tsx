import {render, fireEvent, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {OrganizationSite} from '.'

const sitesReturn = Promise.resolve({
  data: {
    content: [
      {
        id: '750615360979664898',
        createdAt: '2022-04-05T06:27:07.673767Z',
        modifiedAt: '2022-04-08T08:26:25.309229Z',
        version: 5,
        erpId: '102',
        organizationId: '750615355451244546',
        name: 'Ritchie-Kovacekk',
        url: 'http://www.lockmankilbacnn.io',
        status: 'ACTIVE',
      },
    ],
    totalElements: 10,
  },
})

jest.mock('../../services/SitesService', () => {
  const instance = jest.fn().mockReturnValue({
    getSitesByOrganization: jest.fn().mockImplementation(() => sitesReturn),
  })
  return instance
})

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({orgId: 'org-id'}),
}))

test('renders OrganizationSite', async () => {
  const tree = render(
    <MemoryRouter>
      <OrganizationSite />
    </MemoryRouter>
  )
  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
  
  const refresh = tree.container.querySelector(
    '.p-button-icon.pi-refresh'
  ) as Element
  await waitFor(() => {
    expect(refresh).toBeTruthy()
  })

  fireEvent.click(refresh)
  await waitFor(() =>
    expect(tree.queryByText(/Ritchie-Kovacekk/i)).toBeInTheDocument()
  )
  await waitFor(() => {
    expect(
      tree.queryByText(/http:\/\/www.lockmankilbacnn.io/i)
    ).toBeInTheDocument()
  })
})
