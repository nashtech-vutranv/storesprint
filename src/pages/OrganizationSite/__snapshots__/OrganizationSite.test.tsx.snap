// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders OrganizationSite 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="page-title-box"
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                class="breadcrumb m-0"
              >
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="/organizations"
                  >
                    <span>
                      navigator_title_organizations
                    </span>
                  </a>
                </li>
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="/organizations/org-id"
                  >
                    <span>
                       
                      org-id
                       
                    </span>
                  </a>
                </li>
                <li
                  aria-current="page"
                  class="breadcrumb-item active"
                >
                  <span>
                     
                    sites_pate_title
                     
                  </span>
                </li>
              </ol>
            </nav>
            <h4
              class="page-title"
            >
              sites_pate_title
            </h4>
          </div>
        </div>
        <div
          class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh"
          data-scrollselectors=".p-datatable-wrapper"
          pr_id_1=""
        >
          <div
            class="p-datatable-header"
          >
            <div
              class="row"
            >
              <div
                class="d-flex justify-content-start align-items-center"
              >
                <div
                  class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
                >
                  <input
                    class="p-inputtext p-component w-100"
                    placeholder="common_input_search_placeHolder"
                    value=""
                  />
                </div>
                <button
                  class="btn-h-95 btn btn-primary"
                  type="button"
                >
                  common_button_search_label
                </button>
              </div>
            </div>
          </div>
          <div
            class="p-datatable-wrapper"
          >
            <table
              class="p-datatable-table"
              role="table"
            >
              <thead
                class="p-datatable-thead"
              >
                <tr
                  role="row"
                >
                  <th
                    aria-sort="ascending"
                    class="p-sortable-column p-highlight"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        sites_column_header_erpId
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        sites_column_header_name
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="text-break p-sortable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        sites_column_header_url
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        sites_column_header_status
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="p-datatable-tbody"
              >
                <tr
                  class="p-datatable-emptymessage"
                  role="row"
                >
                  <td
                    colspan="4"
                    role="cell"
                  >
                    sites_table_empty_message
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="p-paginator p-component p-paginator-bottom table-paginator"
          >
            <div
              class="p-paginator-left-content"
            >
              <button
                aria-label="refresh-button"
                class="p-button p-component p-button-text p-button-icon-only"
                type="button"
              >
                <span
                  class="p-button-icon p-c pi pi-refresh"
                />
                <span
                  class="p-button-label p-c"
                >
                  Â 
                </span>
              </button>
            </div>
            <button
              class="p-paginator-first p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-2"
              >
                organization_paginator_first
              </span>
            </button>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-3"
              >
                organization_paginator_previous
              </span>
            </button>
            <span
              class="p-paginator-pages"
            />
            <button
              class="p-paginator-next p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-3"
              >
                organization_paginator_next
              </span>
            </button>
            <button
              class="p-paginator-last p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-2"
              >
                organization_paginator_last
              </span>
            </button>
            <div
              class="p-dropdown p-component p-inputwrapper dropdown-table-input opacity-md p-disabled p-inputwrapper-filled"
            >
              <div
                class="p-hidden-accessible"
              >
                <input
                  aria-haspopup="listbox"
                  aria-label="textbox"
                  disabled=""
                  readonly=""
                  type="text"
                />
              </div>
              <div
                class="p-hidden-accessible p-dropdown-hidden-select"
              >
                <select
                  aria-hidden="true"
                  tabindex="-1"
                >
                  <option
                    value=""
                  />
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
              </div>
              <span
                class="p-dropdown-label p-inputtext"
              >
                10
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="dropdown-trigger"
                class="p-dropdown-trigger"
                role="button"
              >
                <span
                  class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
                />
              </div>
            </div>
            <span
              style="user-select: none; text-align: center; margin-left: auto;"
            >
              organization_paginator_show_report
            </span>
            <div
              class="p-paginator-right-content"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="page-title-box"
        >
          <nav
            aria-label="breadcrumb"
          >
            <ol
              class="breadcrumb m-0"
            >
              <li
                class="breadcrumb-item"
              >
                <a
                  href="/organizations"
                >
                  <span>
                    navigator_title_organizations
                  </span>
                </a>
              </li>
              <li
                class="breadcrumb-item"
              >
                <a
                  href="/organizations/org-id"
                >
                  <span>
                     
                    org-id
                     
                  </span>
                </a>
              </li>
              <li
                aria-current="page"
                class="breadcrumb-item active"
              >
                <span>
                   
                  sites_pate_title
                   
                </span>
              </li>
            </ol>
          </nav>
          <h4
            class="page-title"
          >
            sites_pate_title
          </h4>
        </div>
      </div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_1=""
      >
        <div
          class="p-datatable-header"
        >
          <div
            class="row"
          >
            <div
              class="d-flex justify-content-start align-items-center"
            >
              <div
                class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
              >
                <input
                  class="p-inputtext p-component w-100"
                  placeholder="common_input_search_placeHolder"
                  value=""
                />
              </div>
              <button
                class="btn-h-95 btn btn-primary"
                type="button"
              >
                common_button_search_label
              </button>
            </div>
          </div>
        </div>
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  aria-sort="ascending"
                  class="p-sortable-column p-highlight"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      sites_column_header_erpId
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      sites_column_header_name
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="text-break p-sortable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      sites_column_header_url
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      sites_column_header_status
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-datatable-emptymessage"
                role="row"
              >
                <td
                  colspan="4"
                  role="cell"
                >
                  sites_table_empty_message
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom table-paginator"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              aria-label="refresh-button"
              class="p-button p-component p-button-text p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                Â 
              </span>
            </button>
          </div>
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_first
            </span>
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_previous
            </span>
          </button>
          <span
            class="p-paginator-pages"
          />
          <button
            class="p-paginator-next p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_next
            </span>
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_last
            </span>
          </button>
          <div
            class="p-dropdown p-component p-inputwrapper dropdown-table-input opacity-md p-disabled p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                aria-label="textbox"
                disabled=""
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="dropdown-trigger"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
          <span
            style="user-select: none; text-align: center; margin-left: auto;"
          >
            organization_paginator_show_report
          </span>
          <div
            class="p-paginator-right-content"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
